src/main/java/org/gradle/profiler/studio/LauncherConfigurationParser.java

```diff
@@ -0,0 +1,204 @@
+package org.gradle.profiler.studio;
+
+import org.gradle.profiler.instrument.GradleInstrumentation;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class LauncherConfigurationParser {
+    public LaunchConfiguration calculate(Path studioInstallDir) {
+        Path infoFile = studioInstallDir.resolve("Contents/Info.plist");
+        Dict entries = PListParser.parse(infoFile);
+        Dict jvmOptions = entries.dict("JVMOptions");
+        List<Path> classPath = Arrays.stream(jvmOptions.string("ClassPath").split(":")).map(s -> FileSystems.getDefault().getPath(s.replace("$APP_PACKAGE", studioInstallDir.toString()))).collect(Collectors.toList());
+        String mainClass = jvmOptions.string("MainClass");
+        Map<String, String> systemProperties = mapValues(jvmOptions.dict("Properties").toMap(), v -> v.replace("$APP_PACKAGE", studioInstallDir.toString()));
+        Path javaCommand = studioInstallDir.resolve("Contents/jre/jdk/Contents/Home/bin/java");
+        Path agentJar = GradleInstrumentation.unpackPlugin("studio-agent").toPath();
+        Path asmJar = GradleInstrumentation.unpackPlugin("asm").toPath();
+        Path supportJar = GradleInstrumentation.unpackPlugin("instrumentation-support").toPath();
+        Path protocolJar = GradleInstrumentation.unpackPlugin("client-protocol").toPath();
+        return new LaunchConfiguration(javaCommand, classPath, systemProperties, mainClass, agentJar, supportJar, Arrays.asList(asmJar, protocolJar));
+    }
+
+    private static <T, S> Map<String, S> mapValues(Map<String, T> map, Function<T, S> mapper) {
+        Map<String, S> result = new LinkedHashMap<>();
+        for (Map.Entry<String, T> entry : map.entrySet()) {
+            result.put(entry.getKey(), mapper.apply(entry.getValue()));
+        }
+        return result;
+    }
+
+    private static class Dict {
+        private final Map<String, ?> contents;
+
+        public Dict(Map<String, ?> contents) {
+            this.contents = contents;
+        }
+
+        Dict dict(String key) {
+            return (Dict) getEntry(key);
+        }
+
+        String string(String key) {
+            return (String) getEntry(key);
+        }
+
+        Map<String, String> toMap() {
+            return mapValues(contents, v -> (String) v);
+        }
+
+        private Object getEntry(String key) {
+            Object value = contents.get(key);
+            if (value == null) {
+                throw new IllegalArgumentException(String.format("Dictionary does not contain entry '%s'.", key));
+            }
+            return value;
+        }
+    }
+
+    private static class PListParser {
```

Comment:
❓ Have you considered https://github.com/3breadt/dd-plist? I used it in the past and was satisfied with it.

Reply:
No I haven't. I had a quick look for a plist parser and didn't find anything. I'll take a look at using it.